### DISCUSSION TITLE
line height

### DISCUSSION THREAD

**Discussion by bartart3d** [vkbo/novelWriter#653] [Q&A]
Is it possible to change the line spacing between lines? I haven't found any settings for that yet, maybe I'm overlooking something.

If not, it would be a great addition. When I first started trying to write, I used a line spacing of 1 too, as mostly used by text editors, but I met a professional writer, who asked why I don't write, and even print the manuscript in a line spacing of 1.5! She was surprised I did not!

Apparently, most manuscripts are printed with a line spacing of 1.5 when printed on regular size A4 pages (that's here in Belgium the case) because it reads much easier, which is important for the person evaluating your work, and then the amount of pages corresponds more or less to what it would be when published.
She recommended me to do the writing with such line spacing too, which I did in other software, and to be honest, it is much better! It is much easier for the eyes.

grtz,
Bart

**Comment by vkbo** [comment]
Thanks for the suggestion.

I am planning to add a specific export category for manuscripts (#622) which will have a separate set of options in the Build Novel Project tool. Line spacing will definitely be one of those.

Adding a line height option to the main editor is also possible, but [not straight forward](https://stackoverflow.com/questions/10250533/set-line-spacing-in-qtextedit). But I'll look into that as well.

**Reply by vkbo** [reply]
So, I did some testing. For the document viewer, it's trivial. I can just set the line height for the style sheet.

For the editor, Qt has not decided to make line height a global text property of the document. Instead, the only way to apply it is on a per-block basis. That is, the format has to be changed line by line on the document text itself after it is loaded. I can do it in bulk, but it is still slow enough to be in the order of a minute or more to open a large document. This is the same performance bottleneck I've hit before with the PyQt to Qt interface.

For now, I'll make a note of what I did, and perhaps revisit it later. But there are no good solutions to this right now that doesn't break other things. Some of the quirks of Qt are really annoying.

(I will still add this to the manuscript export feature though. The new ODT exporter is pretty much ready in the `dev` branch.)

**Reply by bartart3d** [reply]
Thanks for looking into it so quickly. I'm glad it's not that difficult for the exporter :+1: 
I looked up the link you showed. I'm sure at some point, a solution will be found. Changing line height seems like a common feature in editors, so I hope the Qt-devs will look into at some point.  
I'll look where we can post bug-reports at  Qt, perhaps that might help :-).

**Reply by vkbo** [reply]
Actually, Qt 5.15 is the final version of the Qt5 line, and is an LTS release. They are only releasing patches now. It is very unlikely that they will change this. Especially when they haven't in the last decade either.

Qt 6 is still too early to use for projects like novelWriter. I expect to move to it at some point, but it's likely to be a matter of a couple of years.

But yeah, for exports, it should be available in 1.2. Just need to add an option box for it on the Build dialog.
